### PR TITLE
Create ttt with AI

### PR DESCRIPTION
**Pull request by hscoder**

### CODE REVIEW

File: bin/tictactoe [hunk 1]
```diff
@@ -1,3 +1,30 @@
...
+    elsif input.downcase != "exit"
+      puts "Invalid input. Try again."
+      game_loop
+    else
+      puts "Goodbye."
+      break
+    end
+  end
+
+end
```
Comment (aviflombaum) [inline bin/tictactoe]: Prefer to encapsulate CLI logic into a class. For example, the only code in your bin executable should be:

```ruby
#!/usr/bin/env ruby
       
require_relative '../config/environment'

TicTacToeCLI.new.call
```

And then you'd have a file: `lib/tic_tac_toe_cli.rb` with:

```ruby
class TicTacToeCLI
  def initialize
  end

  def call
   # Your game logic
  end
end
```

The code in your bin would generally just go in `call`.

Comment (aviflombaum) [inline bin/tictactoe]: Additionally even in your example, when is the method you're defining even being evoked?

File: lib/board.rb [hunk 2]
```diff
@@ -0,0 +1,44 @@
+class Board
+  attr_accessor :cells
+
+  def initialize
+    reset! # the #reset! method will run when object gets instantiated and clear the []
+  end
+
+  def reset!
+    @cells = Array.new(9, " ")
```
Comment (aviflombaum) [inline lib/board.rb]: Nice use of `Array.new`

File: bin/tictactoe (hunk 1 above)
Comment (hscoder) [inline bin/tictactoe]: Avi, thanks for helping me keep what's private private with encapsulation. I'll implement the changes. Thx.

File: lib/board.rb (hunk 2 above)
Comment (hscoder) [inline lib/board.rb]: Thx. :grin:

File: lib/game.rb [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+    !won? && full?
+  end
+
+  def full?
+    board.cells.all? { |token| token == "X" || token == "O" } # returns true if elements in cells [] have an "X" or "O"
+  end
+
+  def winner
+    if won?
+      win_combo = won?
```
Comment (randallreedjr) [inline lib/game.rb]: You can accomplish both of these steps in a single line of code.
```
if win_combo = won?
```
If `won?` returns false, the result of the assignment operation will also be false, so the interior of the if statement will not be executed. This pattern is sometimes referred to as 'if assignment.'

**Comment by randallreedjr** [comment]
Great submission @hscoder! I left you one comment in addition to the feedback Avi had already provided. Closing this PR.

**Pull request closed without merging**